src/transaction.ts

```diff
@@ -6,35 +6,53 @@ import {
   TransactionBuilder,
   xdr,
 } from "stellar-base";
+import { SorobanRpc } from "./soroban_rpc";
 
 // TODO: Transaction is immutable, so we need to re-build it here. :(
 export function assembleTransaction(
   raw: Transaction | FeeBumpTransaction,
   networkPassphrase: string,
-  simulated: Array<null | {
-    footprint: Buffer | string | xdr.LedgerFootprint;
-    auth: Array<Buffer | string | xdr.ContractAuth>;
-  }>,
+  simulation: SorobanRpc.SimulateTransactionResponse,
 ): Transaction {
   if ("innerTransaction" in raw) {
     // TODO: Handle feebump transactions
     return assembleTransaction(
       raw.innerTransaction,
       networkPassphrase,
-      simulated,
+      simulation,
     );
   }
 
-  if (simulated.length !== raw.operations.length) {
+  if (
+    raw.operations.length !== 1 ||
+    raw.operations[0].type !== "invokeHostFunction"
+  ) {
     throw new Error(
-      "number of simulated operations not equal to number of transaction operations",
+      "unsupported operation type, must be only one InvokeHostFunctionOp in the transaction.",
+    );
+  }
+
+  const rawInvokeHostFunctionOp: any = raw.operations[0];
+
+  if (
+    !rawInvokeHostFunctionOp.functions ||
+    !simulation.results ||
+    rawInvokeHostFunctionOp.functions.length !== simulation.results.length
+  ) {
+    throw new Error(
+      "preflight simulation results do not contain same count of HostFunctions that InvokeHostFunctionOp in the transaction has.",
     );
   }
 
-  // TODO: Figure out a cleaner way to clone this transaction.
   const source = new Account(raw.source, `${parseInt(raw.sequence, 10) - 1}`);
-  const txn = new TransactionBuilder(source, {
-    fee: raw.fee,
+  const classicFeeNum = parseInt(raw.fee, 10) || 0;
+  const minResourceFeeNum = parseInt(simulation.minResourceFee, 10) || 0;
+  const txnBuilder = new TransactionBuilder(source, {
+    // automatically update the 'classic' tx fee if min resource fees from simulation response
+    // surpass that value.
+    // 'classic' fees are measured as the product of tx.fee * 'number of operations', In soroban contract tx,
```

Comment:
classic fees are `classicFeeNum * operations num` right ?

Reply:
yes, on classic tx's that would be the way classic fees would result, with the limitation of only one operation in the soroban tx, have reduced to just classic fees = classicFeeNum(the parsed numeric value of user specified tx.fee).